test/test_cuda.py

```diff
@@ -3504,6 +3506,62 @@ def throws_on_cuda_event(capture_error_mode):
         # Exception would Corrupt Process and make other tests fail
         # self.assertTrue(throws_on_cuda_event("global"))
 
+    @unittest.skipIf(
+        not TEST_CUDA_GRAPH or not TEST_CUDA_PYTHON,
+        "CUDA >= 11.0 or ROCM >= 5.3 required for graphs, cuda-python must be installed",
+    )
+    @parametrize("instantiate_eagerly", [False, True])
+    def test_cuda_graph_raw_graph(self, instantiate_eagerly):
+        import cuda.bindings.runtime as cudart
+
+        graph = torch.cuda.CUDAGraph(instantiate_eagerly=instantiate_eagerly)
```

Comment:
can we call this arg `keep_graph`, default to False, and eagerly instantiate and destroy graph if it's set to False? That way current memory use will be preserved

Reply:
For my own knowledge, is your concern mostly device memory usage (unsure if `cudaGraph_t` affects this), host-side, or both?

Reply:
To be clear, retaining the cudaGraph_t will increase CPU memory usage, but it won't increase GPU memory usage to the best of my knowledge. cudaGraph_t is fully on CPU, while cudaGraphExec_t is a pointer to a GPU-based data structure. With this knowledge in mind, do you still want me to use the `keep_graph` approach you request here?

Reply:
That's a cool idea, and I think we should add it, but I also think it's good to have explicit control over whether cudaGraph_t is retained.